nmap.go

```diff
@@ -459,6 +485,16 @@ func analyzeWarnings(warnings []string) error {
 	return nil
 }
 
+// WithSudo sets the scanner to run as root using the sudo command.
+func WithSudo() Option {
+	sudoPath, _ := exec.LookPath("sudo")
```

Comment:
I'm not sure if this is a solid way to go. I would prefer passing linux capabilities to the nmap binary.

Reply:
I was thinking about putting that it would only work on *nix machines in the function comment. We could definitely return an Option and an error but then I figured it would break the `With` options design pattern. Is there another way to get my desired result?

I wasn't sure if this PR would be a desired addition or not but I am working on something that would need this capability so I figured I would try to contribute back.

Reply:
First of all I think all contributions are helpful. Also if a PR is not going to be merged. It shows some limitations of this library :) But your need is something we should consider for this wrapper. I used the *capabilities* in a project that is running for two years now and it is working like a charm.

I think it could be a simple idea to include this feature in a new `Run` method.

Maybe we could also build something around this to combine the `Sudo`, `Async`, `WithProgression`, `WithStreamer` methods. Smth like this:
```golang
scanner.Sudo().WithProgression().Async().Run()

//or some new options for running the scanner
Run(options ...RunOption)
```

Reply:
By passing linux _capabilities_ to nmap do you mean setting the different linux capabilities required by nmap on the nmap binary itself like [here](https://secwiki.org/w/Running_nmap_as_an_unprivileged_user)?

Reply:
Yep, exactly. You need to do it once on the system.

Reply:
If linux capabilities is the better way to go, then this PR probably isn't needed. That is, unless you think there could be others who would want this.

Reply:
I think this PR is currently not needed. We could merge it but it could lead to some unexpected issues when the calling user is not in the sudoers list.

Reply:
That is fine with me. I talked with my team and we are ok going the linux capabilities route.

Reply:
Then I will close this PR soon if you don't mind :)

Reply:
That is completely fine with me